packages/plugin-typescript/sources/index.ts

```diff
@@ -27,6 +27,7 @@ const afterWorkspaceDependencyAddition = async (
   const {configuration} = project;
 
   const tsEnableAutoTypes = configuration.get(`tsEnableAutoTypes`)
+    ?? xfs.existsSync(ppath.join(workspace.cwd, `tsconfig.json`))
     ?? xfs.existsSync(ppath.join(project.cwd, `tsconfig.json`));
```

Comment:
```suggestion
    ?? (xfs.existsSync(ppath.join(workspace.cwd, `tsconfig.json`))
    || xfs.existsSync(ppath.join(project.cwd, `tsconfig.json`)));
```
Otherwise it never checks the project root.

Reply:
Thanks, definitely a bug, it was causing https://github.com/yarnpkg/berry/pull/6175#discussion_r1531060132. I'm still not sure if I should re-introduce it though. If you're in a monorepo, and do not have a `tsconfig.json` in one of your workspaces, it might be safe to assume you're not using TypeScript in that specific workspace

Reply:
> If you're in a monorepo, and do not have a tsconfig.json in one of your workspaces, it might be safe to assume you're not using TypeScript in that specific workspace

Not necessarily, this repo is an example of that.

Reply:
Alright thanks, that settles it then, it would have been an odd behavior anyway 👍